aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/waiters/AmazonS3Waiters.java

```diff
@@ -50,7 +50,7 @@ public AmazonS3Waiters(AmazonS3 client) {
      * entered the desired state or not, where polling criteria is bound by
      * either default polling strategy or custom polling strategy.
      */
-    public Waiter bucketNotExists() {
+    public Waiter<HeadBucketRequest> bucketNotExists() {
```

Comment:
Can we all add the generic parameters for the acceptors? (Like `HttpFailureStatusAcceptor<HeadBucketResult>`?)